modeling/02_LS5_out_class_analysis.Rmd

```diff
@@ -0,0 +1,709 @@
+individual labels, rather a list of the scene level cloud cover values
+associated with the AOI.
+
+```{r, echo = FALSE}
+# max
+LS5_out_max_cloud <- outliers %>% 
+  filter(max_cloud_cover > 75)
+# compared with the whole dataset 
+LS5_max_cloud <- LS5_for_class_analysis %>% 
+  filter(class != "cloud" & max_cloud_cover > 75) 
+
+# mean
+LS5_out_mean_cloud <- outliers %>% 
+  filter(mean_cloud_cover > 60) 
+# compared with the whole dataset 
+LS5_mean_cloud <- LS5_for_class_analysis %>% 
+  filter(class != "cloud" & mean_cloud_cover > 60) 
+```
+
+The outlier dataset contains `r nrow(LS5_out_max_cloud)`
+(`r round(nrow(LS5_out_max_cloud)/nrow(outliers)*100, 1)`%) where the max cloud
+cover was \> 75% and `r nrow(LS5_out_mean_cloud)`
+(`r round(nrow(LS5_out_mean_cloud)/nrow(outliers)*100, 1)`%) where the mean
+cloud cover was \> 50%. The filtered dataset contains `r nrow(LS5_max_cloud)`
+(`r round(nrow(LS5_max_cloud)/nrow(LS5_for_class_analysis)*100, 1)`%) where max
+was \>75% and `r nrow(LS5_mean_cloud)`
+(`r round(nrow(LS5_mean_cloud)/nrow(LS5_for_class_analysis)*100, 1)`%) where the
+mean cloud cover was \> 50%. Neither of these metrics seem helpful in elimination
+of labels from outlier scenes.
+
+### Training dataset implications
+
+Based on the above review, any label with SR_ATMOS_OPACITY value >= 0.3, or a QA
+flag for clouds, cloud shadow, or snow ice should be eliminated from this dataset.
+
+```{r}
+LS5_training_labels <- LS5_for_class_analysis %>% 
+  filter(class == "cloud" |
+           (SR_ATMOS_OPACITY < 0.3 &
+            cloud_conf < 2 &
+            cloudshad_conf <2 & 
+            snowice_conf <2))
+```
+
+## Testing for inter-class differences
+
+We do want to have an idea of how different the classes are, in regards to band
+data. While there are a bunch of interactions that we could get into here, for
+the sake of this analysis, we are going to analyze the class differences by
+band.
+
+Kruskal-Wallis assumptions:
+
+1)  Data are non-Normal or have a skewed distribution
+2)  There must be at least two independent groups.
+3)  Data have a similar distribution across groups.
+4)  Data are independent, the groups shouldn't have a relationship to one
+    another
+5)  Each group should contain at least 5 observations
+
+ANOVA assumptions:
+
+1)  data are distributed normally
+2)  data are independent
+3)  variance across groups is similar
+
+We can't entirely assert sample independence and we know that variance and
+distribution is different for "cloud" labels, but those data also are visibly
+different from the other classes.
+
+In order to systematically test for differences between classes and be able to
+intepret the data, we will need to know some things about our data:
+
+1)  Are the data normally distributed (Shapiro-Wilkes)?
+2)  Are there outliers that may impact interpretation?
+3)  If data is non-normal, perform Kruskal-Walis test; otherwise ANOVA
```

Comment:
great point! I'll throw that in there - Since all the data are non-normal/skewed, I might just add a print of 'Kruskal-Wallis is not the right test!' instead of going through the ANOVA steps.

Reply:
And ended up just adding another test and if/else nest in here.